sfcasts/ep1/data-providers.md

```diff
@@ -1,42 +1,42 @@
 # Data Providers
 
-We treat our source code as first-class citizens. Why not do the same with our
+We treat our source code as a first-class citizen. That means, among other things,
+we avoid duplication. Why not do the same with our
 tests? Our three tests for the size are... repetitive. They test the same thing
-just with slightly different input and then a different assertion. We can improve
-this by using PHPUnit's Data Providers - which run the same test over and over
-with different sets of data.
+just with *slightly* different input and then a different assertion. Is there
+a way to improve this? Absolutely: thanks to PHPUnit Data Providers.
 
 ## Refactor our tests 
 
-Move down to the bottom of our `DinosaurTest` and add
+Move to the bottom of `DinosaurTest` and add
 `public function sizeDescriptionProvider()`. Inside, `yield` an array with `[10, 'Large']`,
 then `yield [5, 'Medium']`, and finally `yield [4, 'Small']`. Yield is just a fancy
-way of returning arrays using PHP's built-in Generator function.
+way of returning arrays using PHP's built-in Generator function. As you'll see in
+a minute, these keys - like `10` and `large` will become *arguments* to our test.
```

Comment:
My philosophy: "Show first, explain later". But, in this case, I felt that we needed a BIT more explanation so the user can kind of understand the purpose of the provider method and why we're now adding arguments to the test method.